chore(docker): containerize repo (#351)
algolia/search-bundle: DOCKER_README.MD

In this page you will find our recommended way of installing Docker on your machine. 
This guide is made for OSX users.

> ## Install docker

Reviewer: Can you apply the suggested changes in the other repos on this README, please? E.g. `Docker` is not capitalized everywhere in this file. Thank you 😄

First install Docker using [Homebrew](https://brew.sh/)
```
$ brew install docker
```

You can then install [Docker Desktop](https://docs.docker.com/get-docker/) if you wish, or use `docker-machine`. As we prefer the second option, we will only document this one.

## Setup your Docker

Install `docker-machine`
```
$ brew install docker-machine
```

Then install [VirtualBox](https://www.virtualbox.org/) with [Homebrew Cask](https://github.com/Homebrew/homebrew-cask) to get a driver for your Docker machine
```
$ brew cask install virtualbox
```

You may need to enter your password and authorize the application in your `System Settings` > `Security & Privacy`.

> Create now a new machine, set it up as default and connect your shell to it (here we use zsh. The commands should anyway be displayed in each steps' output)

Reviewer: ```suggestion
Create a new machine, set it up as default and connect your shell to it.
```

```
$ docker-machine create --driver virtualbox default
$ docker-machine env default
$ eval "$(docker-machine env default)"
```

> Now you're all setup to use our provided Docker image!

Reviewer: ```suggestion
Now you're all set to use our Docker image!
```

## Build the image

```bash
docker build -t algolia-search-bundle .
```

## Run the image

> You need to provide few environment variables at runtime to be able to run the [Common Test Suite](https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite).

Reviewer: ```suggestion
You need to provide a few environment variables at runtime in order to run the [Common Test Suite](https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite).
```

You can set them up directly in the command:

```bash
docker run -it --rm --env ALGOLIA_APP_ID=XXXXXX [...] -v $PWD:/app -w /app algolia-search-bundle bash
```

However, we advise you to export them in your `.bashrc` or `.zshrc`. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to set your variables.

```bash
docker run -it --rm --env ALGOLIA_APP_ID \
                    --env ALGOLIA_API_KEY \
-v $PWD:/app -w /app algolia-search-bundle bash
```

Once your container is running, any changes you make in your IDE are directly reflected in the container.

> Manually install composer dependencies once inside the container

Reviewer: ```suggestion
The first time you start the container, you have to manually install the composer dependencies.
```

Create now a new machine, set it up as default and connect your shell to it (here we use zsh. The commands should anyway be displayed in each steps' output)

```
$ docker-machine create --driver virtualbox default
$ docker-machine env default
$ eval "$(docker-machine env default)"
```

Now you're all setup to use our provided Docker image!

## Build the image

```bash
docker build -t algolia-search-bundle .
```

## Run the image

You need to provide few environment variables at runtime to be able to run the [Common Test Suite](https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite).
You can set them up directly in the command

```bash
docker run -it --rm --env ALGOLIA_APP_ID=XXXXXX [...] -v $PWD:/app -w /app algolia-search-bundle bash
```

But we advise you export them in your `.bashrc` or `.zshrc`. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to retrieve your variables.

```bash
### For external contributors, the following env variables should be enough

> docker run -it --rm --env ALGOLIA_APP_ID \

Reviewer: I ran this command after the setup, but it didn't work; I had to manually run `composer install` before it worked. This was my output:

```
composer test
Do not run Composer as root/super user! See https://getcomposer.org/root for details
> php-cs-fixer fix -v --dry-run
sh: 1: php-cs-fixer: not found
Script php-cs-fixer fix -v --dry-run handling the test:lint event returned with error code 127
Script @test:lint was called via test
```

It looks like the `RUN composer install` in the `Dockerfile` doesn't do as expected. Any idea why?

Author: @DevinCodes could you try again? it may related to the fact I didn't put the `vendor/` folder in the `.dockerignore` file

Reviewer: I still have the same issue :(
